storagext/lib/src/clients/market.rs

```diff
@@ -360,6 +373,37 @@ impl MarketClientExt for crate::runtime::client::Client {
             .await
     }
 
+    async fn retrieve_all_deal_parameters(
+        &self,
+    ) -> Result<
+        Vec<(
+            <crate::PolkaStorageConfig as subxt::Config>::AccountId,
+            SpecializedRuntimeDealParameters,
+        )>,
+        subxt::Error,
+    > {
+        let deal_parameters_query = runtime::storage().market().sp_deal_parameters_iter();
+
+        let mut deal_params = self
+            .client
+            .storage()
+            .at_latest()
+            .await?
+            .iter(deal_parameters_query)
+            .await?;
+
+        let mut params = vec![];
+
+        while let Some(Ok(kv)) = deal_params.next().await {
+            let bytes = &kv.key_bytes[(kv.key_bytes.len() - 32)..];
+            let array_u8: [u8; 32] = bytes.try_into().expect("On-chain validation of accountIDs");
+            let account = <crate::PolkaStorageConfig as subxt::Config>::AccountId::from(array_u8);
+            params.push((account, kv.value))
```

Comment:
[Look ma', no `expect`! (hidden panic though)](https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice)

This is the approach I suggested on the call:
```suggestion
            let mut buffer = [0u8; 32];
            buffer.copy_from_slice(&kv.key_bytes[(kv.key_bytes.len() - 32)..]);
            let account = <crate::PolkaStorageConfig as subxt::Config>::AccountId::from(buffer);
            params.push((account, kv.value))
```